### ISSUE TITLE
Enforce permissive license on all federated content to protect instance admins

### ISSUE THREAD

**Issue by skopf** [mastodon/mastodon#23201]
### Pitch

Include a section in the standard TOS applying a permissive license like CC0 or CC-BY to all content uploaded by users. Encourage instance admins to adopt these terms in their custom TOS.
Instance admins may use a different license for "instance only" posts, but not for "public" posts.

**Related issues:**
There are multiple issues asking for the ability to amend content with a license:

- #22931
- #9528
- #20079

However, these would make the current situation even more dangerous for instance admins, for example if someone chooses a more restrictive license like CC-BY-NC. This would make it impossible for commercial actors like mastodon.social to display the content.

### Motivation

Currently, Mastodon's default TOS do not mention copyright at all. This is a dangerous situation for instance admins. The problem lies in the federation of content.
A toot provided by a user to one instance is replicated to other instances and then displayed on their websites. This happens on multiple pages, like the federated timeline, a tag page or a profile page (if the toot got boosted, replied to or is a reply itself).

**There needs to be a clear legal basis on which instances are using federated content.**

Currently, there is a severe risk that a gold digger could exploit instance admins as follows:
Create multiple accounts on different instances and follow each other. Then make a public toot with one of the accounts and optionally attach a picture. Wait for the toot to be federated to the other instances and to appear on their federated timelines. Then take legal action against the other instances for displaying your copyrighted material.
It will be quite tricky for the instance owner to argue against this complaint. The gold digger has only given consent to the home instance to display the content publicly.  Especially the section

> We do not sell, trade, or otherwise transfer to outside parties your personally identifiable information.

in the privacy policy makes it hard to argue that the gold digger has given consent to the use of the copyrighted material outside of the home instance.
Instances which host a follower might be able to argue that the content has been pushed _to_ them. But instances only receiving the boost of the toot are actively going back to the home instance and download the toot and picture from there.

From multiple comments on the above mentioned feature requests you can see that users are not aware that they need to license their content under a permissive license to be able to participate in the Fediverse. With the massive influx of new users and professional content creators like journalists and artists, it is IMHO just a question of _when_ the s**t will hit the fan. There will be situations where commercial instances will display content where the creator is not happy with this usage and is taking legal action.

**The spacial situation with federation**
In a centralised network operated by a single entity, like Twitter, Flickr or Instagram, it is sufficient to license the content to this single entity for a clearly defined use case (e.g. displaying on their website).
The Fediverse however is a federation of countless independent instances operated by different entities. At the time of writing a public toot, it is impossible to know or limit on which instances it will be displayed. It is important that any entity, commercial or non-commercial, has the right to use this toot for their website.
It is even impossible to limit the use cases (e.g. only for federated timelines). There are multiple software projects out there in the fediverse all connected by ActivityPub using the federated content in very different ways. For example, some are removing additional formatting from the HTML content and are thus creating a "derivative work".

**Comment by nightpool** [comment]
The fediverse has operated for over a decade on the implicit understanding that content is licensed by the poster for distribution to their followers, since that's the obvious and clearly foreseeable outcome of using the software. I don't think any court would see a user uploading their images onto the account that they signed up for, and then seeing that those images were distributed to the people who followed them, as anything except the explicit performance of the actions they undertook. We're working to augment that implicit understand with an explicit, purpose-limited grant of the rights necessary for participation in the social network with a new terms of service. However, we believe a purpose limitation is crucially important here, because if a user happens to upload art or content that they're working on to Mastodon, we don't believe that they're automatically surrendering all rights to use that content, just as they're free to upload e.g. in progress storyboards for an animated short to Twitter without worrying that somebody else might take those storyboards and turn them into a film. Users should feel free to participate in the network without worrying about jeopardizing their livelihoods with a too-broad license. For that reason, we will not adopt a CC0 of CC-BY requirement for posting to Mastodon. Similarly, a non-commercial restriction like CC-BY-NC would raise concerns about instance hosting costs and funding which would be too strict in the opposite direction. For that reason, we think a terms of service that codifies the existing informal understanding with an explicit _purpose-limited_ grant is the best path forward.

**Comment by skopf** [comment]
Absolutely agree with everything you said. I've only suggested CC styled licenses since they are easy to apply. If you are working on a specific purpose-limited licenses, that's even better.
Feel free to close this issue right away if you are already working on it, or do you want to keep it open until the new TOS are released?

**Comment by shleeable** [comment]
I agree that images should have licenses attached, but what do you mean by dangerous situation?

**Comment by nemobis** [comment]
Cc @mindspillage. :innocent:

**Comment by supernovae** [comment]
I don't care what copyright a user claims if they signal that on their post, but i don't think Mastodon should be in the license and enforcement business of any license.   We should just have a strong UGR.  The last thing I want to see is more moderator actions for license violation/enforcement.

**Comment by bkil** [comment]
Related issue: https://github.com/mastodon/mastodon/issues/10123
